File: Ex2.cs
using System;
using System.Collections.Generic;

class Program
{
    // Функция для преобразования римского числа в натуральное число
    static int RomanToInt(string romanNumeral)
    {
        // Создаем словарь для соответствия римских цифр и их значений
        var romanNumerals = new Dictionary<char, int>
        {
            {'I', 1},
            {'V', 5},
            {'X', 10},
            {'L', 50},
            {'C', 100},
            {'D', 500},
            {'M', 1000}
        };

        int result = 0;
        int prev = 0;
        for (int i = romanNumeral.Length - 1; i >= 0; i--)
        {
            int current = romanNumerals[romanNumeral[i]];
            if (current < prev)
            {
                result -= current;
            }
            else
            {
                result += current;
            }
            prev = current;
        }

        return result;
    }

    static void Main(string[] args)
    {
        // Получаем ввод от пользователя
        Console.Write("Введите римское число: ");
        string input = Console.ReadLine();

        // Преобразуем римское число в натуральное и выводим результат
        int result = RomanToInt(input.ToUpper()); // Преобразуем в верхний регистр для удобства
        Console.WriteLine("Результат: " + result);
    }
}

File: Ex2.js
// Функция для преобразования римского числа в натуральное число
function romanToInt(romanNumeral) {
    const romanNumerals = { 'I': 1, 'V': 5, 'X': 10, 'L': 50, 'C': 100, 'D': 500, 'M': 1000 };
    let result = 0;
    for (let i = 0; i < romanNumeral.length; i++) {
        const current = romanNumerals[romanNumeral[i]];
        const next = romanNumerals[romanNumeral[i + 1]];
        if (next && current < next) {
            result += next - current;
            i++;
        } else {
            result += current;
        }
    }
    return result;
}

// Получаем ввод от пользователя и выводим результат
const romanNumeral = prompt("Введите римское число:");
console.log("Результат:", romanToInt(romanNumeral));
